artifactory/commands/transferfiles/utils.go

```diff
@@ -118,17 +120,17 @@ func pollUploads(phaseBase *phaseBase, srcUpService *srcUserPluginService, uploa
 		if progressbar != nil {
 			progressbar.SetRunningThreads(curProcessedUploadChunks)
 		}
-		curTokensBatch.fillTokensBatch(uploadTokensChan)
-
-		if len(curTokensBatch.UuidTokens) == 0 {
+		fillTokensBatch(awaitingStatusChunksSet, uploadTokensChan)
+		if awaitingStatusChunksSet.Size() == 0 && len(curTokensBatch.ChunksToDelete) == 0 {
```

Comment:
Let's explain here to the random reader what `awaitingStatusChunksSet` stores and also what `ChunksToDelete`, and then what does it mean when the size for both is zero.

Reply:
done